### ISSUE TITLE
Messages with the same type name are not resolved correctly

### ISSUE THREAD

**Issue by fferri** [CoppeliaRobotics/v_repExtRosInterface#9]
_From @leokoppel on November 5, 2016 3:56_

I have some message types which happen to have the same name, in different packages. I add them to meta/messages.txt:

```
a_msgs/Foo
a_msgs/Bar
b_msgs/Foo
```

Also, the message description of `a_msgs/Bar` includes the relative type name `Foo`.

Building produces errors of the form:
```
.../generated/ros_msg_io.cpp: In function ‘void read__a_msgs__Bar(int, a_msgs::Bar*, const ReadOptions*)’:
.../generated/ros_msg_io.cpp:8125:75: error: cannot convert ‘a_msgs::Bar_<std::allocator<void> >::_cmd_type* {aka a_msgs::Foo_<std::allocator<void> >*}’ to ‘b_msgs::Foo* {aka b_msgs::Foo_<std::allocator<void> >*}’ for argument ‘2’ to ‘void read__b_msgs__Foo(int, b_msgs::Foo*, const ReadOptions*)’
                         read__b_msgs__Foo(stack, &(msg->cmd), opt);
```

among other errors.

The problem is relative type names are not resolved correctly. They should be resolved according to ROS'  [Message Description Specification](http://wiki.ros.org/msg#Message_Description_Specification):

> When embedding other Message descriptions, the type name may be relative (e.g. "Point32") if it is in the same package; otherwise it must be the full Message type (e.g. "std_msgs/String"). The only exception to this rule is Header. 

Therefore a relative type name must be interpreted as a Message in the same package. However, `generate_ros_stuff.py` neglects this in the `resolve_msg` dict, allowing the relative type name `Foo` inside `a_msgs/Bar` to be incorrectly resolved as `b_msgs/Foo`.

Luckily, I think the solution is simple: `rosmsg show`, *without* the "raw" option, will properly resolve type names itself.

_Copied from original issue: fferri/v_repExtRosInterface#9_

**Comment by fferri** [comment]
_From @leokoppel on November 5, 2016 4:11_

My proposed fix builds on top of existing PR #8, so I'll hold off on making another PR.

**Comment by fferri** [comment]
Thanks for your contribution. Please make a PR for this too.
